File: fixtures/java/type_inference/cross_file/Service.java
package app.service;

import app.models.User;
import app.models.UserFactory;

public class Service {
    public String renderGreeting(User user) {
        return user.greet();
    }

    public void main() {
        User user = UserFactory.createUser("Alice");
        System.out.println(renderGreeting(user));
        System.out.println(user.greet());
    }
}

File: fixtures/scala/type_inference/cross_file/Service.scala
package app.service

import app.models.{User, UserFactory}

object Service {
  def renderGreeting(user: User): String = user.greet()

  def main(args: Array[String]): Unit = {
    val user = UserFactory.createUser("Alice")
    println(renderGreeting(user))
    println(user.greet())
  }
}
